backend/test/integration/run_api_test.go

```diff
@@ -264,7 +270,11 @@ func (s *RunApiTestSuite) checkTerminatedRunDetail(t *testing.T, runDetail *run_
 		ScheduledAt: runDetail.Run.ScheduledAt,
 		FinishedAt:  runDetail.Run.FinishedAt,
 	}
-	assert.True(t, reflect.DeepEqual(expectedRun, runDetail.Run))
+
+	// Need to sort resource references before equality check as the order is non-deterministic
+	sort.Sort(ResourceReferenceSorter(runDetail.Run.ResourceReferences))
+	sort.Sort(ResourceReferenceSorter(expectedRun.ResourceReferences))
+	assert.Equal(t, expectedRun, runDetail.Run)
```

Comment:
@chensun will https://godoc.org/github.com/stretchr/testify/assert#ElementsMatch be a simpler solution?

Reply:
We've been using it in https://github.com/kubeflow/pipelines/blob/152c91c705911239a71ff9400c40805799cfd85f/backend/test/integration/job_api_test.go#L356 for exactly the same reason

Reply:
I personally think the need to set resource reference to null after `ElementsMatch` check is bit hacky. But keeping code consistent is more important in this case. So I'm okay to adopt the already used solution.

Reply:
Looks like we only used `ElementsMatch` once in our entire project. And with that pattern, we need to define expectedResourceReferences outside expected{Run|Job}, do the resource references ElementsMatch check first, then set actual{Run|Job}.ResourcesReferences to nil, and do the final comparison. In comparison, I think sorting before one comparison is just easier to follow. So I'm leaning towards the other way -- changing that `ElementsMatch` to sorting. WDTY?
-----
A side note, I would leave this change as-is as it's now pending for retest+merge, and others are waiting for this to unblock presubmit. 
I will send a separate PR to make code consistent, one way or another.

Reply:
Yes, there is only one case. Your arguments sound reasonable. We can reuse the sort solution.